crypto/elgamal/ciphertext.go

```diff
@@ -58,51 +59,46 @@ func (z *Ciphertext) Add(x, y *Ciphertext) *Ciphertext {
 }
 
 // Serialize returns a slice of len 4*32 bytes,
-// representing the C1.X, C1.Y, C2.X, C2.Y as little-endian.
+// representing the C1.X, C1.Y, C2.X, C2.Y as little-endian,
+// in reduced twisted edwards form.
 func (z *Ciphertext) Serialize() []byte {
-	x1, y1 := z.C1.Point()
-	x2, y2 := z.C2.Point()
 	var buf bytes.Buffer
-	for _, bi := range []*big.Int{
-		x1,
-		y1,
-		x2,
-		y2,
-	} {
+	c1x, c1y := format.FromTEtoRTE(z.C1.Point())
+	c2x, c2y := format.FromTEtoRTE(z.C2.Point())
+	for _, bi := range []*big.Int{c1x, c1y, c2x, c2y} {
 		if _, err := buf.Write(arbo.BigIntToBytes(32, bi)); err != nil {
 			panic(err)
 		}
 	}
 	return buf.Bytes()
 }
 
-// Deserialize reconstructs a Ciphertext from a slice of bytes.
-// The input must be of len 4*32 bytes, representing the C1.X, C1.Y, C2.X, C2.Y as little-endian.
-func (z *Ciphertext) Deserialize(data []byte) error {
+// Deserialize reconstructs an Ciphertext from a slice of bytes.
+// The input must be of len 4*32 bytes (otherwise it panics),
+// representing the C1.X, C1.Y, C2.X, C2.Y as little-endian,
+// in reduced twisted edwards form.
+func (z *Ciphertext) Deserialize(data []byte) {
 	const fieldSize = 32 // Each field element is 32 bytes
 	expectedLen := 4 * fieldSize
 
 	// Validate the input length
 	if len(data) != expectedLen {
-		return fmt.Errorf("invalid input length: got %d bytes, expected %d bytes", len(data), expectedLen)
+		panic(fmt.Errorf("invalid input length: got %d bytes, expected %d bytes", len(data), expectedLen))
```

Comment:
Please don't Panic here... it's something that might happen. Return an error.

Reply:
my intention was to keep in line with Serialize, that never returns an error and can panic during buf.Write
// If the buffer becomes too large, Write will panic with [ErrTooLarge].

but no big deal, i'll return the error instead, LGTM